File: Backend/src/main/java/com/example/Xtrend_Ai/dto/PodcastResponse.java
package com.example.Xtrend_Ai.dto;

import com.example.Xtrend_Ai.enums.Status;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import java.net.URL;
import java.util.List;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Setter
@JsonInclude(JsonInclude.Include.NON_NULL)
public class PodcastResponse {


    /// use this response to allow client to poll the backend

    private Long podcastId;
    private String key;
    private Status status;

    private String url;

    private String title;
    private String description;
    private String imageUrl;
    private List<String> country;
    private List<String>category;

    private String podcastTagline;



}

File: Frontend/src/app/models/PodcastResponse.ts
export interface PodcastResponse{
    podcastId:number;
    key?:string;
    status?:string;

    url?:string

    imageUrl?:string
    description?:string
    country?:Array<string>
    category?:Array<string>

    podcastTagline?:string
    createdAt?:string

}